conf/configuration.go

```diff
@@ -203,9 +229,48 @@ func MinPinners(ctx context.Context, db *database.DB) (int, error) {
 		return 0, err
 	}
 	if mp < minPinnersMinValue || mp > maxPinnersMinValue {
-		errMsg := fmt.Sprintf("Invalid min_pinners value in database configuration! The value must be between %d and %d, it was %v.", mp, minPinnersMinValue, maxPinnersMinValue)
+		errMsg := fmt.Sprintf("invalid min_pinners value in database configuration! The value must be between %d and %d, it was %v.", mp, minPinnersMinValue, maxPinnersMinValue)
 		build.Critical(errMsg)
 		return 0, errors.New(errMsg)
 	}
 	return int(mp), nil
 }
+
+// NextScan returns the time of the next cluster-wide scan for underpinned files.
+func NextScan(ctx context.Context, db *database.DB, logger logger.Logger) (time.Time, error) {
+	val, err := db.ConfigValue(ctx, ConfNextScan)
+	if errors.Contains(err, mongo.ErrNoDocuments) {
+		logger.Infof("Missing database value for '%s', setting a new one.", ConfNextScan)
+		// No scan has been scheduled. Schedule one in an hour.
+		err = SetNextScan(ctx, db, time.Now().Add(DefaultNextScanOffset).UTC())
+		if err != nil {
+			return time.Time{}, err
+		}
+		// Read the value from the DB in order to avoid race conditions.
+		// While this does not eliminate data races, it does help.
+		return NextScan(ctx, db, logger)
+	}
+	if err != nil {
+		return time.Time{}, err
+	}
+	t, err := time.Parse(time.RFC3339, val)
+	if err != nil {
+		logger.Warnf("Invalid database value for '%s': '%s', setting a new one.", ConfNextScan, val)
+		err = SetNextScan(ctx, db, time.Now().Add(DefaultNextScanOffset).UTC())
+		if err != nil {
+			return time.Time{}, err
+		}
+		// Read the value from the DB in order to avoid race conditions.
```

Comment:
As I said in https://github.com/SkynetLabs/pinner/pull/41#discussion_r934246666, I don't think this data race is important in any way. It can only happen if:
1. we are just deploying this functionality for the first time on that cluster
2. we have two servers which have different builds
3. we have changed the value of `DefaultNextScanOffset` between those builds

Even if all those are somehow met, the scan also needs to be within the `SleepBetweenChecksForScan` period for both servers in order to cause an actual discrepancy. And even if all those thing happen, the end result will still be that one server start scanning for underpinned skylinks a bit before the other. That will have zero effect on a brand new cluster and a very small effect on an existing cluster because we intentionally pause between skylink repins.

All in all, I don't think it's worth it to add the extra complexity and DB load of using locks. I changed this code to 
```go
		scanTime := time.Now().Add(DefaultNextScanOffset).UTC()
		err = SetNextScan(ctx, db, scanTime)
		if err != nil {
			return time.Time{}, err
		}
		return scanTime, nil
```
based on your previous comment and I think that's the right call.

Reply:
oh yeah as long as we get rid of the comment that mentions the race condition as well as the redundant fetch I'm totally fine, thanks!